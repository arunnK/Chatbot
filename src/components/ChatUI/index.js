import React from 'react';
import { Toast } from 'react-bootstrap';
import PropTypes from 'prop-types';

const ChatUI = ({ serverMessages, clientMessages }) => {
  return (
    <React.Fragment>
      {serverMessages.map((serverMessage, index) => (
        <React.Fragment>
          <span>You: </span>
          <Toast show style={{ width: '50%' }}>
            <Toast.Body>{clientMessages[index]}</Toast.Body>
          </Toast>
          <span style={{ width: '50%', marginLeft: '55%' }}>Bot: </span>
          <Toast show style={{ width: '50%', marginLeft: '55%' }}>
            <Toast.Body>{serverMessage}</Toast.Body>
          </Toast>
        </React.Fragment>
      ))}
    </React.Fragment>
  );
};

ChatUI.propTypes = {
  serverMessages: PropTypes.arrayOf(PropTypes.string).isRequired,
  clientMessages: PropTypes.arrayOf(PropTypes.string).isRequired
};

ChatUI.defaultProps = {};

export default ChatUI;
