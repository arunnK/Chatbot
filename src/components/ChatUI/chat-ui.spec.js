import { create, act } from 'react-test-renderer';
import React from 'react';
import ChatUI from './index';
// in your test:
const serverMessages = ['hey', 'hi'];
const clientMessages = ['hey', 'hi'];

let root;

test('chat page', () => {
  act(() => {
    root = create(<ChatUI serverMessages={serverMessages} clientMessages={clientMessages} />);
  });
  // make assertions on root
  expect(root.toJSON()).toMatchSnapshot();
});
