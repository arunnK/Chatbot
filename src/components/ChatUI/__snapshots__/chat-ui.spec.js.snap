// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
Array [
  <div
    aria-atomic="true"
    aria-live="assertive"
    className="fade toast show"
    role="alert"
    style={
      Object {
        "width": "50%",
      }
    }
  >
    <div
      className="toast-body"
    >
      hey
    </div>
  </div>,
  <div
    aria-atomic="true"
    aria-live="assertive"
    className="fade toast show"
    role="alert"
    style={
      Object {
        "marginLeft": "55%",
        "width": "50%",
      }
    }
  >
    <div
      className="toast-body"
    >
      hey
    </div>
  </div>,
  <div
    aria-atomic="true"
    aria-live="assertive"
    className="fade toast show"
    role="alert"
    style={
      Object {
        "width": "50%",
      }
    }
  >
    <div
      className="toast-body"
    >
      hi
    </div>
  </div>,
  <div
    aria-atomic="true"
    aria-live="assertive"
    className="fade toast show"
    role="alert"
    style={
      Object {
        "marginLeft": "55%",
        "width": "50%",
      }
    }
  >
    <div
      className="toast-body"
    >
      hi
    </div>
  </div>,
]
`;

exports[`chat page 1`] = `
Array [
  <div
    aria-atomic="true"
    aria-live="assertive"
    className="fade toast show"
    role="alert"
    style={
      Object {
        "width": "50%",
      }
    }
  >
    <div
      className="toast-body"
    >
      hey
    </div>
  </div>,
  <div
    aria-atomic="true"
    aria-live="assertive"
    className="fade toast show"
    role="alert"
    style={
      Object {
        "marginLeft": "55%",
        "width": "50%",
      }
    }
  >
    <div
      className="toast-body"
    >
      hey
    </div>
  </div>,
  <div
    aria-atomic="true"
    aria-live="assertive"
    className="fade toast show"
    role="alert"
    style={
      Object {
        "width": "50%",
      }
    }
  >
    <div
      className="toast-body"
    >
      hi
    </div>
  </div>,
  <div
    aria-atomic="true"
    aria-live="assertive"
    className="fade toast show"
    role="alert"
    style={
      Object {
        "marginLeft": "55%",
        "width": "50%",
      }
    }
  >
    <div
      className="toast-body"
    >
      hi
    </div>
  </div>,
]
`;

exports[`mock hook 1`] = `
Array [
  <div
    aria-atomic="true"
    aria-live="assertive"
    className="fade toast show"
    role="alert"
    style={
      Object {
        "width": "50%",
      }
    }
  >
    <div
      className="toast-body"
    >
      hey
    </div>
  </div>,
  <div
    aria-atomic="true"
    aria-live="assertive"
    className="fade toast show"
    role="alert"
    style={
      Object {
        "marginLeft": "55%",
        "width": "50%",
      }
    }
  >
    <div
      className="toast-body"
    >
      hey
    </div>
  </div>,
  <div
    aria-atomic="true"
    aria-live="assertive"
    className="fade toast show"
    role="alert"
    style={
      Object {
        "width": "50%",
      }
    }
  >
    <div
      className="toast-body"
    >
      hi
    </div>
  </div>,
  <div
    aria-atomic="true"
    aria-live="assertive"
    className="fade toast show"
    role="alert"
    style={
      Object {
        "marginLeft": "55%",
        "width": "50%",
      }
    }
  >
    <div
      className="toast-body"
    >
      hi
    </div>
  </div>,
]
`;
